### ISSUE TITLE
workspace.asRelativePath returns nonsensical value

### ISSUE THREAD

**Issue by DanTup** [Microsoft/vscode#10446]
- VSCode Version: 1.4
- OS Version: Windows 10

The docs for `workspace.asRelativePath` state:

> When there is no workspace.rootPath or when the path is not a child of that folder, the input is returned.

However, I'm running this code:

	let input = "m:\\Apps\\DartPubCache\\hosted\\pub.dartlang.org\\convert-2.0.1\\lib\\src\\hex.dart";

	console.log("root: " + workspace.rootPath);
	console.log("input: " + input);
	console.log("output: " + workspace.asRelativePath(input));

and seeing this output:

    root: m:\Coding\Applications\NewsWoWBot
    input: m:\Apps\DartPubCache\hosted\pub.dartlang.org\convert-2.0.1\lib\src\hex.dart
    output: artlang.org\convert-2.0.1\lib\src\hex.dart

Now, possibly confusing things, the input path does actually have a symlink pointing to it that is inside the workspace (at `{workspaceRoot}\packages\convert\src\hex.dart` is the link) however I don't believe that should result in a malformed path.

**Comment by jrieken** [comment]
It's actually just some substring logic...

**Comment by DanTup** [comment]
heh, I converted a `startsWith` to use a wrapper over this (`isInProject`) to avoid any complications if I had overlapping names and there wasn't a trailing slash!

I just found [this code](https://github.com/Microsoft/vscode/blob/581c260f9a72c4c7e6f2a6ca92596fa27e4c0b3a/src/vs/workbench/api/node/extHostWorkspace.ts#L40) but it assume if the path is longer it's inside; doesn't actually compare the value - I presume that's the issue?

I'll change my wrapper to compare manually for now. Thanks!
